analyzers/src/SonarAnalyzer.Common/SymbolicExecution/Roslyn/SymbolicCheck.cs

```diff
@@ -25,6 +25,9 @@ public class SymbolicCheck
         protected SymbolicCheck() { } // Avoid abstract class, fixes S1694
 
         public virtual ProgramState PreProcess(SymbolicContext context) => context.State;
+
         public virtual ProgramState PostProcess(SymbolicContext context) => context.State;
+
+        public virtual ProgramState ExitReached(ProgramState state) => state;
```

Comment:
We will need more than just the context.
We don't need to return a new state, because we don't expect any cross-check modifications there. And we don't use it.
```suggestion
        public virtual void ExitReached(SymbolicContext context) { }
```

Reply:
If you can think of a scenario where `ProgramState` is needed, we can use this
```suggestion
        public virtual ProgramState ExitReached(SymbolicContext context) => context.State;
```
and then invoke it like this from the engine:
```
InvokeChecks(context, x => x.ExitReached)
```

Reply:
Currently, the `SymbolicContext` has only the State and the Operation. When this part is reached the operation is always null. I prefer to keep this as simple as possible and extend it only when needed.

Reply:
The context will pretty soon contain more information, especially about the flows. Please give it the `SymbolicContext context` to keep the pattern the same for all.